src/Tests/dotnet.Tests/dotnet-msbuild/GivenDotnetBuildInvocation.cs

```diff
@@ -84,7 +88,46 @@ public void MsbuildInvocationIsCorrect(string[] args, string expectedAdditionalA
                     .Should()
                     .Be($"{ExpectedPrefix} -nologo -consoleloggerparameters:Summary{expectedAdditionalArgs}");
             });
+        }
+
+        [Fact]
+        public void OsOptionIsCorrectlyResolved()
+        {
+            CommandDirectoryContext.PerformActionWithBasePath(WorkingDirectory, () =>
+            {
+                var msbuildPath = "<msbuildpath>";
+                var command = BuildCommand.FromArgs(new string[] { "--os", "os" }, msbuildPath);
+                var expectedArch = Environment.Is64BitOperatingSystem ? "x64" : "x86";
+                command.GetArgumentsToMSBuild()
+                    .Should()
+                    .Be($"{ExpectedPrefix} -restore -consoleloggerparameters:Summary -property:RuntimeIdentifier=os-{expectedArch}");
+            });
+        }
+
+        [WindowsOnlyFact]
+        public void ArchOptionIsCorrectlyResolved()
+        {
+            CommandDirectoryContext.PerformActionWithBasePath(WorkingDirectory, () =>
+            {
+                var msbuildPath = "<msbuildpath>";
+                var command = BuildCommand.FromArgs(new string[] { "--arch", "arch" }, msbuildPath);
+                command.GetArgumentsToMSBuild()
+                    .Should()
+                    .Be($"{ExpectedPrefix} -restore -consoleloggerparameters:Summary -property:RuntimeIdentifier=win-arch");
+            });
+        }
 
+        [Fact]
+        public void OSAndArchOptionsCanBeCombined()
+        {
+            CommandDirectoryContext.PerformActionWithBasePath(WorkingDirectory, () =>
+            {
+                var msbuildPath = "<msbuildpath>";
+                var command = BuildCommand.FromArgs(new string[] { "--arch", "arch", "--os", "os" }, msbuildPath);
+                command.GetArgumentsToMSBuild()
+                    .Should()
+                    .Be($"{ExpectedPrefix} -restore -consoleloggerparameters:Summary -property:RuntimeIdentifier=os-arch");
+            });
         }
```

Comment:
Other tests to add:
* Specify the same option twice (not sure if the command line parsing handles this implicitly) - `dotnet build -a x86 -a x64`
* Test on all the other commands - not just build - at least test that both options are accepted and that they work - can we make this into a matrix test instead?
* Test what happens if the new options are combined with `-r`
* Test what happens if the project specifies `RuntimeIdentifier` as a property in `.csproj` - which one will win?

Reply:
The first item is covered with command line parsing implicitly and the last item is implicitly handled by MSBuild (since command line args are global properties). But I'll add tests for the other two, thanks.